docs/control-plane-architecture.adoc

```diff
@@ -0,0 +1,252 @@
+
+Broadly, the control plane must provide:
+
+* an externally-facing API endpoint described in <<rfd4>> through which users can provision elastic infrastructure backed by the system.  This includes APIs for compute Instances, storage, networking, as well as supporting resources like organizations, users, groups, ssh keys, tags, and so on.  This API may be used by developers directly as well as the developer console backend.  See <<rfd30>>.
+* an externally-facing API endpoint for all operator functions.  This is a long list, including configuration and management of hardware and software components and monitoring.
+* implementation of lifecycle activities, like initial system setup; adding, removing, or replacing servers or other components; and the like.
+* facilities for remote support by Oxide, including secure access to crash dumps, core files, log files, and system consoles.
+
+== Fundamental properties
+
+**Availability.**  Availability of the control plane refers to the property that requests to provision resources succeed when the underlying resources are available within the system and requests to reconfigure or monitor resources succeed as long as they are well-formed.  Unavailability refers to request failure due to hardware or software failure.
+
+IMPORTANT: Generally, the control plane is expected to remain **available** in the face of any two hardware or software failures, including transient failures of individual compute sleds, power rectifiers, switches, or the like.
+
+**Durability.**  Along the same lines, resources created in the control plane are expected to be durable unless otherwise specified.  That is, if the whole system is powered off and on again ("cold start"), the system should converge to a point where all Instances, disks, and networking resources that were running before the power outage are available as they were from the user's perspective before the event.  Similarly, if a compute server is lost (either through graceful decommissioning or otherwise), it should be possible to resume service of resources that were running on that server (e.g., Instances, disks) on other servers in the system.  There may be additional constraints on how many servers can fail permanently before data is lost, but in no case should it be possible to permanently lose an Instance, disk, or other resource after the permanent failure of two compute sleds.
```

Comment:
I noticed that my instances on rack3 (colo) were powered off after upgrade to R9.  Was this the expected behavior?

Reply:
Yes. We currently do not have online upgrade.

Reply:
I've added a note to this section that this is not all implemented yet.